asg.tf

```diff
@@ -131,6 +131,12 @@ module "asg" {
     http_put_response_hop_limit = var.disable_container_credentials ? 1 : 2
   }
 
+  suspended_processes = [
+    # Prevents the ASG from terminating instances for rebalancing between AZs, 
+    # which triggered right after termination of instances by lambda
+    "AZRebalance"
+  ]
+
```

Comment:
I think that should be optional. Someone could use asg without `autoscaler`, then `AZRebalance` could be helpful.

Reply:
How? I don't see options for this in this module

Reply:
Or you mean disabling of Lambda autoscaler?
https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?tab=readme-ov-file#input_enable_autoscaling

Reply:
If so - there is solution
https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/96#pullrequestreview-2265973382

Also, there can be provided `length(var.suspended_processes) > 0 ? var.suspended_processes : var.enable_autoscaling ? ["AZRebalance"] : []`

Reply:
or:
```diff
--- a/asg.tf
+++ b/asg.tf
@@ -131,11 +131,7 @@ module "asg" {
     http_put_response_hop_limit = var.disable_container_credentials ? 1 : 2
   }
 
-  suspended_processes = [
-    # Prevents the ASG from terminating instances for rebalancing between AZs, 
-    # which triggered right after termination of instances by lambda
-    "AZRebalance"
-  ]
+  suspended_processes = var.asg_suspended_processes
 
   # User data
   user_data = base64encode(
diff --git a/variables.tf b/variables.tf
index 9af4b6b..537789e 100644
--- a/variables.tf
+++ b/variables.tf
@@ -226,3 +226,8 @@ variable "autoscaler_s3_package" {
   default     = null
 }
 
+variable "asg_suspended_processes" {
+  type = list(string)
+  description = "List of processes to suspend. For example, `AZRebalance` if you're using https://github.com/spacelift-io/ec2-workerpool-autoscaler - this will prevent the ASG from terminating instances for rebalancing between AZs, which triggered right after termination of instances by lambda."
+  default = null
+}
```

Reply:
Then we can place it there something else if we need to.